src/devices/Bmxx80/Bmx280Base.cs

```diff
@@ -0,0 +1,198 @@
+    /// </summary>
+    public abstract class Bmx280Base : Bmxx80Base
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Bmx280Base"/> class.
+        /// </summary>
+        /// <param name="i2cDevice">The <see cref="I2cDevice"/> to create with.</param>
+        protected Bmx280Base(byte deviceId, I2cDevice i2cDevice)
+            : base(deviceId, i2cDevice)
+        {
+            _calibrationData = new Bmx280CalibrationData();
+            _calibrationData.ReadFromDevice(this);
+            _controlRegister = (byte)Bmx280Register.CONTROL;
+        }
+
+        /// <summary>
+        /// Read the temperature.
+        /// </summary>
+        /// <returns>Calculated temperature.</returns>
+        public async Task<Temperature> ReadTemperatureAsync()
+        {
+            if (ReadPowerMode() == Bmx280PowerMode.Forced)
+            {
+                await Task.Delay(GetMeasurementTimeForForcedMode(ReadTemperatureSampling()));
+            }
+
+            //Read the MSB, LSB and bits 7:4 (XLSB) of the temperature from the BMP280 registers
+            byte msb = Read8BitsFromRegister((byte)Bmx280Register.TEMPDATA_MSB);
+            byte lsb = Read8BitsFromRegister((byte)Bmx280Register.TEMPDATA_LSB);
+            byte xlsb = Read8BitsFromRegister((byte)Bmx280Register.TEMPDATA_XLSB); // bits 7:4
+
+            //Combine the values into a 32-bit integer
+            int t = (msb << 12) + (lsb << 4) + (xlsb >> 4);
+
+            return CompensateTemperature(t);
+        }
+
+        /// <summary>
+        /// Recommended wait timings from the datasheet.
+        /// </summary>
+        /// <param name="sampleMode">The <see cref="Sampling"/> to get for.</param>
+        /// <returns>The time it takes for the chip to read data in milliseconds rounded up.</returns>
+        private int GetMeasurementTimeForForcedMode(Sampling sampleMode)
+        {
+            if (sampleMode == Sampling.UltraLowPower)
+            {
+                return 7;
+            }
+            else if (sampleMode == Sampling.LowPower)
+            {
+                return 9;
+            }
+            else if (sampleMode == Sampling.Standard)
+            {
+                return 14;
+            }
+            else if (sampleMode == Sampling.HighResolution)
+            {
+                return 23;
+            }
+            else if (sampleMode == Sampling.UltraHighResolution)
+            {
+                return 44;
+            }
+            return 0;
+        }
+
+        /// <summary>
+        /// Read the <see cref="Bmx280PowerMode"/> state.
+        /// </summary>
+        /// <returns>The current <see cref="Bmx280PowerMode"/>.</returns>
+        public Bmx280PowerMode ReadPowerMode()
+        {
+            byte read = Read8BitsFromRegister(_controlRegister);
+
+            // Get only the power mode bits.
+            var powerMode = (byte)(read & 0b_0000_0011);
+
+            return (Bmx280PowerMode)powerMode;
```

Comment:
NIT: Looks like PowerMode only supports 0b00 and 0b01 so we should validate that what we got is not 0b10 or 0b11.

Reply:
Would the expected behavior be at that point to throw an exception? e.g.
```csharp
if (Enum.IsDefined(typeof(Bmx280PowerMode), powerMode) == false)
{
    throw new IOException("Read unexpected power mode");
}
```